tests/base_tests.py

```diff
@@ -511,24 +542,46 @@ def test_fit_not_changing_data_weighted(
     ):
         """Test fit does not change X - when weights are used."""
         df = minimal_dataframe_lookup[self.transformer_name]
+        uninitialized_transformer = uninitialized_transformers[self.transformer_name]
+
+        # skip polars test if not narwhalified
+        if not uninitialized_transformer.polars_compatible and isinstance(
+            df,
+            pl.DataFrame,
+        ):
+            return
+
+        df = nw.from_native(df)
```

Comment:
is this to get around the issue you were having with narwhalified methods in super calls?